public/src/overmind/state.tsx

```diff
@@ -226,6 +230,19 @@ export const state: State = {
     users: {},
     allUsers: [],
     courses: [],
+    courseTeachers: derived(({ courseEnrollments, activeCourse }: State) => {
+        if (!activeCourse || !courseEnrollments[activeCourse.toString()]) {
+            return {}
+        }
+        const teachers = courseEnrollments[activeCourse.toString()].filter(isTeacher)
```

Comment:
Here you iterate through all users enrolled and then iterate again through the teachers. Wouldn’t it be better to just iterate once over all enrolled users and check if it is a teacher and directly add it to the teachersMap?

Reply:
Fixed.

Reply:
What I meant is that `filter(isTeacher)` is actually an iteration over `courseEnrollments`... My proposal was to do:
```ts
for (const enrolled of courseEnrollments[activeCourse.toString()]) {
  if (isTeacher(enrolled)) {
...
```

Reply:
Appears I forgot to actually push the fix before I left the office.. 🤡 The "fix" I refer to does what you propose, except to use 
```ts
courseEnrollments[activeCourse.toString()].forEach(enrollment => {
    ...
})

Reply:
sounds good